### PR TITLE
Default the manifest path to app/packs/manifest.json

### PR DESCRIPTION
**Pull request by jdelStrother**
This makes the manifest path configurable, as well as defaulting it to app/packs/manifest.json

Previously discussed here: https://github.com/rails/webpacker/issues/2876

Not sure if app/packs is the best place for it, I'm open to other suggestions.

I wondered if we could get away with changing the default to app/packs since I _think_ by default it will continue to work with many people's setups. I think the main issue would be if, eg, someone is running `assets:precompile` on a CI server and tarring up public/packs to ship off to their production servers, they'll no longer have the manifest.json included in that tar file. But for anyone that just runs `assets:precompile` on all their production servers as part of a deploy, they won't notice much difference unless they're actually trying to load manifest.json out of their public path for some reason.

If we were able to get this into the 6.0 release that would be amazing because it would mean we don't have to argue about changing the default behaviour, but I know that 6.0 has already got RCs out so maybe it's a bit late.

One issue is that you can end up overwriting manifest.json with the test environment's version of it - prior to this, they'd be kept separate in public/packs/manifest.json & public/packs-test/manifest.json.  In my own app I tweak the test environment to avoid this:

```yaml
test:
  <<: *default
  compile: true

  # Compile test packs to a separate directory
  public_output_path: packs-test
  manifest_path: app/packs/manifest-test.json
```

Possibly that should be included in the default config.

### CODE REVIEW

**Comment by jdelStrother** [comment]
If I fix the merge conflicts is there anything I can do to help get this merged, or is it a wontfix?

**Comment by justin808** [comment]
@jdelStrother I'm not clear on why you would want the packs-test (for rails testing) manifest to be in the main packs (development and production) directory.

Is this a real security concern to be able to see the manifest file?

and if so, then it would make more sense to just move the location of the manifest to from /public/packs to /manifests/packs and /manifests/packs-test.

I'm not convinced that we need this.

@guillaumebriday I recommend that we close and reopen if a real security issue.

**Comment by jdelStrother** [comment]
> I'm not clear on why you would want the packs-test (for rails testing) manifest to be in the main packs (development and production) directory.

I'm not sure I follow..? It's in RAILS_ROOT/app/packs, which is shared between all environments.

> Is this a real security concern to be able to see the manifest file?

TBD, I guess. In my opinion it's leaking information that might not be desirable to be leaked, but see https://github.com/rails/webpacker/issues/2876 for more discussion.

> and if so, then it would make more sense to just move the location of the manifest to from /public/packs to /manifests/packs and /manifests/packs-test.

As in RAILS_ROOT/manifests/? Are you suggesting introducing a new top-level directory, or is there other stuff in the rails ecosystem that writes to RAILS_ROOT/manifests/ ?

File: lib/install/config/webpacker.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@
 default: &default
   source_path: app/javascript
   source_entry_path: /
+  manifest_path: app/javascript/manifest.json
```
Comment (justin808) [inline lib/install/config/webpacker.yml:6]: I think it's awkward to include a build-generated, not in git, file inside of `/app`

File: lib/install/template.rb [hunk 2]
```diff
@@ -18,12 +18,13 @@
...
 if File.exists?(git_ignore_path)
   append_to_file git_ignore_path do
-    "\n"                   +
-    "/public/packs\n"      +
-    "/public/packs-test\n" +
-    "/node_modules\n"      +
-    "/yarn-error.log\n"    +
-    "yarn-debug.log*\n"    +
+    "\n"                              +
+    "/app/javascript/manifest.json\n" +
```
Comment (justin808) [inline lib/install/template.rb:22]: Per prior comment.

**Review by justin808 — CHANGES_REQUESTED** [review]
I think having the manifest path be configurable is reasonable.

However, I'd have the default stay as the current location, rather than changing the default.

Why? Many build processes cache the webpack generated files and putting the output off somewhere else adds (unnecessary) complexity.

This PR is also missing a changelog entry and documentation.

**Pull request closed without merging**